### PR TITLE
k8s: integration tests use roles instead of cluster roles, events watcher and uid watcher respect namespace [ch2686]

### PR DESCRIPTION
**Pull request by maiamcc**
Hello tbd,

Please review the following commits I made in branch maiamcc/watch-permissions-for-role:

a77b82679269585a8cb6463e71e7d5c2bbf389c9 (2019-06-06 11:29:41 -0400)
k8s: integration tests use roles instead of cluster roles, events watcher and uid watcher respect namespace [ch2686]

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/k8s/watch.go [hunk 1]
```diff
@@ -317,16 +337,24 @@ func (kCli K8sClient) WatchEverything(ctx context.Context, lps []model.LabelPair
 	}
 
 	var watchers []watch.Interface
+	fmt.Printf("🔎 found %d watchable gvrs\n", len(gvrs))
 	for _, gvr := range gvrs {
+		fmt.Println("👀 making watcher for", gvr.String())
```
Comment (jazzdan) [inline internal/k8s/watch.go]: 👀

File: internal/k8s/watch.go [hunk 2]
```diff
@@ -317,16 +337,24 @@ func (kCli K8sClient) WatchEverything(ctx context.Context, lps []model.LabelPair
 	}
 
 	var watchers []watch.Interface
+	fmt.Printf("🔎 found %d watchable gvrs\n", len(gvrs))
```
Comment (jazzdan) [inline internal/k8s/watch.go]: 👀

File: internal/k8s/watch.go [hunk 3]
```diff
@@ -30,9 +30,19 @@ type watcher interface {
 	Watch(options metav1.ListOptions) (watch.Interface, error)
 }
 
+type gvrWithNamespaced struct {
+	schema.GroupVersionResource
+	namespaced bool // if true, this resource can exist within a namespace; if false, cluster-wide
```
Comment (jazzdan) [inline internal/k8s/watch.go]: if the comment goes above the struct member then certain tools/IDEs will pull that comment in when you hover over a reference to it.

File: internal/k8s/watch.go (hunk 1 above)
Comment (maiamcc) [reply internal/k8s/watch.go]: that's uh. that's supposed to be there!

File: internal/k8s/watch.go (hunk 2 above)
Comment (maiamcc) [reply internal/k8s/watch.go]: ALSO SUPPOSED TO BE THERE WHAT ARE YOU TALKING ABOUT

**Comment by maiamcc** [comment]
well integration tests may not run right locally for me, but looks like i can manifest the fix locally! (left is master, right is this branch)
Screen Shot 2019-06-06 at 12 55 30 PM

**Pull request merged**